Plugin/Magento/Framework/View/Element/UiComponent/DataProvider/RegularFilterPlugin.php

```diff
@@ -58,20 +64,96 @@ public function aroundApply(
         Collection $collection,
         Filter $filter
     ) {
+        $filterValue = $filter->getValue();
         if ($collection instanceof OrderGridCollection &&
-            $filter->getField() == 'payment_method' &&
-            ! ($filterValue = $filter->getValue()) &&
-            strpos(is_array($filterValue) ? implode(',', $filterValue) : (string)$filterValue, Payment::METHOD_CODE . '_') !== false
+            $filter->getField() == self::PAYMENT_METHOD_FILTER_CODE &&
+            $filterValue &&
+            $this->isFilterHasBoltPaymentMethod($filterValue)
         ) {
             $collectionSelect = $collection->getSelect();
             $collectionAdapter = $collectionSelect->getConnection();
-            $paymentMethod = str_replace(Payment::METHOD_CODE . '_', '', $filter->getValue());
-            $collectionSelect->where(
-                'main_table.payment_method = "'. Payment::METHOD_CODE .'"
-                AND ('. $collectionAdapter->quoteInto('LOWER(payment.additional_data) like ?', '%' . $paymentMethod . '%') .' OR '. $collectionAdapter->quoteInto('LOWER(payment.additional_information) like ?', '%' . $paymentMethod . '%') .' OR '. $collectionAdapter->quoteInto('LOWER(payment.cc_type) = ?', $paymentMethod) .')'
-            );
+            $paymentMethods = $this->getPaymentMethodsWithTypes($filterValue);
+            $where = '';
+            foreach ($paymentMethods as $paymentMethod) {
+                //if where statement already have some statement all next should be as OR
+                if ($where) {
+                    $where .= ' OR ';
+                }
+                if ($paymentMethod['type'] == self::BOLT_PAYMENT_METHOD_TYPE) {
+                    //bolt payment method where statement
+                    $where .= '(main_table.payment_method = "'. Payment::METHOD_CODE .'"
+            AND ('. $collectionAdapter->quoteInto('LOWER(payment.additional_data) like ?', '%' . $paymentMethod['code'] . '%') .' OR '. $collectionAdapter->quoteInto('LOWER(payment.additional_information) like ?', '%' . $paymentMethod['code'] . '%') .' OR '. $collectionAdapter->quoteInto('LOWER(payment.cc_type) = ?', $paymentMethod['code']) .'))';
+                } else {
+                    //default payment method where statement
+                    $where .= '(main_table.payment_method = "'. $paymentMethod['code'] .'")';
```

Comment:
This is fine but let's also skip the plugin completely if there is no bolt payment methods at all.

Reply:
already there